sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -121,8 +123,11 @@ class SessionCatalog(
     currentDb = db
   }
 
-  def getDefaultDBPath(db: String): String = {
-    System.getProperty("java.io.tmpdir") + File.separator + db + ".db"
+  def createDatabasePath(dbName: String, path: Option[String]): String = {
+    val dbPath = path.map(new Path(_)).getOrElse {
+      new Path(new Path(System.getProperty("java.io.tmpdir")), dbName.toLowerCase() + ".db")
+    }
+    dbPath.toString.stripSuffix(File.separator)
```

Comment:
This is to remove the last `File.separator`, if existing. Based on the Hive behavior, the location does not have it, not matter whether the original user-input location has it or not.